api/v1alpha1/api.go

```diff
@@ -144,3 +144,74 @@ const (
 	// This can be used to describe the routing behavior in HTTPRoute referenced by LLMRoute.
 	LLMModelHeaderKey = "x-envoy-ai-gateway-llm-model"
 )
+
+// LLMProviderType specifies the type of the LLMProviderPolicy.
+type LLMProviderType string
+
+const (
+	LLMProviderTypeAPIKey LLMProviderType = "APIKey"
+)
+
+// +kubebuilder:object:root=true
+
+// LLMProviderPolicy specifies the provider specific configuration.
+type LLMProviderPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              LLMProviderPolicySpec `json:"spec,omitempty"`
+}
+
+// LLMProviderPolicySpec specifies a provider (e.g.AWS Bedrock, Azure etc.) specific-configuration like auth
+type LLMProviderPolicySpec struct {
+	// BackendRefs lists the LLMBackends that this provider policy will apply
+	// The namespace is "local", i.e. the same namespace as the LLMRoute.
+	//
+	BackendRefs []egv1a1.BackendRef `json:"backendRefs,omitempty"`
```

Comment:
Shouldn't this ref to LLMBackend? so we can remove providerType field as well.

Reply:
@mathetake do you have advice here? I saw that you removed the [LLMBackendRef in this commit](https://github.com/envoyproxy/ai-gateway/commit/8299827281dd4b92f26a3881744ca7687cca3fa1#diff-788b9806fbf2a26408470898083b955fff036495d42d54707301ed2d38137ad0L60-L63)

Reply:
Going to update this so LLMBackend points to SecurityPolicy

Reply:
Could you clarify why we should remove the providerType?

Reply:
yeah to be honest the existence of "Type" is just a convention following the EG APIs (e.g. `type` field of [HealthCheck](https://gateway.envoyproxy.io/latest/api/extension_types/#activehealthcheck).) For me personally I would rather remove it (making everything optional is enough).

Reply:
btw I like the current form where LLMBackend referencing LLMSecurityPolicy rather than the other way around